src/privatesend/privatesend.h

```diff
@@ -53,27 +53,64 @@ enum PoolMessage {
     MSG_SUCCESS,
     MSG_ENTRIES_ADDED,
     ERR_SIZE_MISMATCH,
-    MSG_POOL_MIN = ERR_ALREADY_HAVE,
-    MSG_POOL_MAX = ERR_SIZE_MISMATCH
 };
+template<> struct is_serializable_enum<PoolMessage> : std::true_type {};
 
 // pool states
-enum PoolState {
+enum PoolState : int32_t {
     POOL_STATE_IDLE,
     POOL_STATE_QUEUE,
     POOL_STATE_ACCEPTING_ENTRIES,
     POOL_STATE_SIGNING,
     POOL_STATE_ERROR,
-    POOL_STATE_SUCCESS,
-    POOL_STATE_MIN = POOL_STATE_IDLE,
```

Comment:
Please note that the serialization code does not check for valid enum values...it does a plain and stupid cast to the enum type. So, validity/range checks must still be done (e.g. as part of SerializationOp)